### ISSUE TITLE
Passing function to params breaks nested populates

### ISSUE THREAD

**Issue by kshitizshankar** [marshallswain/feathers-graph-populate#16]
Passing a function to params breaks the nested populates it seems. Not sure how to preserve the $populateParams when passing a function to params in the relationship which utilizes context to create dynamic params.

For example - if a relationship uses a param function 
```
    params: (params, context) => {
          params.scopeId= context.params.scopeId
        }
```
and has a nested query that needs to be populated
```
query: {
    posts: {
        users: {}
   }
}
```

Think it happens due to the way $populateParams are stored on the populates' before being passed to shallow-populate - which breaks when the params in the populate definition is a function.
https://github.com/marshallswain/feathers-graph-populate/blob/dc4602b15471180d86761374a3f36c48d4a06186/lib/graph-populate.js#L31

This results in $populateParams being lost when the shallow-populate runs the param function which in turn results in all nested queries not getting executed. 
https://github.com/Mattchewone/feathers-shallow-populate/blob/374dd961b47e2bc35dac86fdc609fa76e5096606/lib/shallow-populate.js#L115

Not really sure to resolve this (still trying to figure out what exactly is happening here)

**Comment by kshitizshankar** [comment]
Creating a wrapper around the params function in the graph-populate seems to work without changing anything else in the flow.

```
const wrapper = function(currentQuery, paramsFunction) {
  return (params, context) => {
    paramsFunction(params, context)
    Object.assign(params, {
      $populateParams: {
        query: currentQuery
      }
    })
  }
}
```
and assigning that wrapper to `data.params` when params is a function
```
if(isFunction(data.params)) {
  data.params = wrapper(currentQuery, data.params)
}
else {
  Object.assign(data.params, {
    $populateParams: {
      query: currentQuery
    }
  })
}
```

Not sure if this would introduce any other issues or not (haven't tested it out properly).

**Comment by fratzinger** [comment]
Hey :)
I think the problem lies in the Line 31 you mentioned above. I will have a look into it. I change some things in `feathers-shallow-populate` right now. After that I will come to this.

**Comment by fratzinger** [comment]
It will be fixed with #20 which heavily relies on https://github.com/Mattchewone/feathers-shallow-populate/pull/19
